src/react/RegisterDevicePage.js

```diff
@@ -0,0 +1,241 @@
+        <Form onSubmit={this.registerDevice}>
+          <Form.Row>
+            <Form.Group as={Col} controlId="deviceIdInput">
+              <Form.Label>Device ID</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="Your device ID"
+                value={this.state.deviceId}
+                onChange={this.updateDeviceId}
+                autoComplete="off"
+                required
+                isValid={this.state.isValidDeviceId}
+                isInvalid={this.state.isInvalidDeviceId}
+              />
+              <Form.Control.Feedback type="invalid">
+                The device ID must be 22 characters long and only contain
+                letters, numbers and the following characters _ -
+              </Form.Control.Feedback>
+            </Form.Group>
+          </Form.Row>
+          <Form.Row className="flex-row-reverse pr-2">
+            <Button
+              variant="primary"
+              type="submit"
+              disabled={!this.state.isValidDeviceId}
+            >
+              Register device
+            </Button>
+            <Button
+              variant="secondary"
+              className="mx-1"
+              onClick={this.generateRandomUUID}
+            >
+              Generate random ID
+            </Button>
+          </Form.Row>
+        </Form>
+        {this.renderModal()}
+      </SingleCardPage>
+    );
+  }
+
+  registerDevice(e) {
+    e.preventDefault();
+
+    this.astarte
+      .registerDevice(this.state.deviceId)
+      .then(this.handleRegistrationSuccess)
+      .catch(this.handleRegistrationError);
+  }
+
+  handleRegistrationSuccess(response) {
+    let secret = response.data["credentials_secret"];
+
+    this.setState({
+      showModal: true,
+      modalTitle: "Device Registered!",
+      modalBody: (
+        <>
+          <span>The device credential secret is</span>
+          <pre className="my-2">
+            <code
+              id="secret-code"
+              className="m-1 p-2 bg-light"
+              style={{ fontSize: "1.2em" }}
+            >
+              {secret}
+            </code>
+            <i
+              className="fas fa-paste"
+              onClick={pasteSecret}
+              style={{ cursor: "copy" }}
+            />
+          </pre>
+          <span>
+            Please don't share the Credentials Secret, and ensure it is
+            transferred securely to your Device.
+            <br />
+            Once the Device pairs for the first time, the Credentials Secret
+            will be associated permanently to the Device and it won't be
+            changeable anymore.
+          </span>
+        </>
+      )
+    });
+  }
+
+  handleRegistrationError(err) {
+    const { name, message } = err;
+
+    this.setState({
+      showModal: true,
+      modalTitle: name,
+      modalBody: <p>{message}</p>
+    });
+  }
+
+  setNewDeviceId(deviceId) {
+    const isValid = deviceId.length == 22;
```

Comment:
it is valid when it can be decoded from base64 (url encoded), and decoded size is 16 bytes.